src/components/mypage/UserContent.tsx

```diff
@@ -1,25 +1,36 @@
 import styled from '@emotion/styled';
 import { useQuery } from '@tanstack/react-query';
 
-import RecommendationPost from '../common/RecommendationPost';
-import { getPostFeedLimit } from './api';
+import BattleCard from '@/components/battle/Card';
+import RecommendationPost from '@/components/common/RecommendationPost';
+
+import { getBattlesLimit, getPostFeedLimit } from './api';
 import ContentList from './ContentList';
 
 function UserContent() {
   const { data: postFeedLimit } = useQuery(['postfeedlimit'], getPostFeedLimit);
+  const { data: battleLimit } = useQuery(['battlelimit'], getBattlesLimit);
+
+  const battleList = battleLimit?.map((battle) => ({
+    id: battle.battleId,
+    isProgress: battle.battleStatus === 'PROGRESS',
+    challenging: {
+      albumCoverImage: battle.challenging.albumUrl,
+      title: battle.challenging.title,
+      singer: battle.challenging.singer,
+    },
+    challenged: {
+      albumCoverImage: battle.challenged.albumUrl,
+      title: battle.challenged.title,
+      singer: battle.challenged.singer,
+    },
+  }));
```

Comment:
오직 컴포넌트에 넣어주기 위한 변환이니까 그냥 이렇게 해주어도 좋을 것 같아요~

```suggestion
  const BattleList = battleLimit?.map((battle) => (
    <BattleCard
      id={battle.id}
      challenging={
        {
          albumCoverImage: battle.challenging.albumUrl,
          title: battle.challenging.title,
          singer: battle.challenging.singer,
      }
    },
    challenged = {
        {
          albumCoverImage: battle. challenged.albumUrl,
          title: battle. challenged.title,
          singer: battle. challenged.singer,
      }
    }
    isProgress={battle.battleStatus === 'PROGRESS'} />
  )) ;
```

Reply:
ask;
변환 작업을 하신 이유가 challenging, challenged의 nickname 필드 및 genre 필드가 필요 없기 때문인가요?

Reply:
@Park-min-hyoung API에서 받아온 데이터와 컴포넌트 내에서 사용할 데이터 타입이 달라서 매핑 시켜주는 작업이라고 해요!

Reply:
@NamJwong 코멘트 참고해서 구조분해할당으로 수정했습니당~

Reply:
아 그렇군요~ 감사합니다!